packages/shared/src/components/Feed.tsx

```diff
@@ -283,25 +284,42 @@ export default function Feed<T>({
       return item.post.id === openedPost;
     });
 
-    if (index === items.length) {
-      await fetchPage();
+    if (index === items.length - 1) {
+      if (isFetchingNextPage) {
+        return null;
+      }
+
+      setIsFetchingNextPage(true);
+      return fetchPage();
     }
 
+    if (items[index + 1].type === 'placeholder') {
+      return null;
+    }
+
+    setIsFetchingNextPage(false);
     const offset = items[index + 1].type !== 'post' ? 2 : 1;
     const item = items[index + offset];
 
     if (item?.type !== 'post') {
```

Comment:
This will cause a weird behavior that if I hit next and there's an ad the modal will close itself right?

Reply:
No. This was just required because of typescript. But handling the ad will be at line 301 where it decides how much should be the offset.